### PR TITLE
Use `config_dict_or_path` for deepspeed.zero.Init

### PR DESCRIPTION
**Pull request by aphedges**
# What does this PR do?

While #13587 fixes one DeepSpeed warning, it missed a similar one. This applies the same change of `config` -> `config_dict_or_path` in another location.

## Who can review?

@sgugger (same reviewer as #13587)

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for fixing!

**Comment by sgugger** [comment]
I'll let @stas00 merge this after he double-checks it's good :-)

**Review by stas00 — APPROVED** [review]
Thank you for catching this omission, @aphedges!

**Comment by aphedges** [comment]
@stas00, you're welcome! Thanks for the review!

**Pull request merged**
